### PR TITLE
Support adding permissions through the Java client

### LINKED ISSUE #22147: Support adding permissions in the Java Client
<h2>Description</h2>

Send an Authorization ADD_PERMISSION command via the Java Client. This also includes integration testing of the feature.

Blocked by #21778

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

Adds the possibility to add permissions to an authorization in the Java Client

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #22147

### CODE REVIEW

**Comment by remcowesterhoud** [comment]
@Ben-Sheppard FYI if this one is merged you can use the Java client to add permissions to a user.

**Review by nicpuppa — CHANGES_REQUESTED** [review]
Great work @remcowesterhoud 💪 

🔧 I left only a small suggestion 

❌ QA Integration tests are missing

**Comment by remcowesterhoud** [comment]
> ❌ QA Integration tests are missing

Can't believe I forgot 🤦 I found some things I could improve when writing these so I added those as well.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -2075,6 +2079,12 @@ components:
           type: "string"
         email:
           type: "string"
+    UserCreateResponse:
+      type: "object"
+      properties:
+        userKey:
```
Comment (nicpuppa) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2085]: 🔧 Description is missing

File: service/src/main/java/io/camunda/service/UserServices.java [hunk 2]
```diff
@@ -43,8 +44,8 @@ public UserServices withAuthentication(final Authentication authentication) {
     return new UserServices(brokerClient, searchClient, transformers, authentication);
   }
 
-  public CompletableFuture<UserRecord> createUser(final CreateUserRequest request) {
-    return sendBrokerRequest(
+  public CompletableFuture<BrokerResponse<UserRecord>> createUser(final CreateUserRequest request) {
+    return sendBrokerRequestWithFullResponse(
```
Comment (nicpuppa) [inline service/src/main/java/io/camunda/service/UserServices.java:48]: ❓ Why this change ? In `ResponseMapper` you are using only the response object, and not the full `BrokerResponse`. Am I missing something ?

File: zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerAuthorizationCreateRequest.java [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class BrokerAuthorizationCreateRequest extends BrokerExecuteCommand<Autho
   private final AuthorizationRecord requestDto = new AuthorizationRecord();
 
   public BrokerAuthorizationCreateRequest() {
-    super(ValueType.AUTHORIZATION, AuthorizationIntent.CREATE);
+    super(ValueType.AUTHORIZATION, AuthorizationIntent.ADD_PERMISSION);
```
Comment (nicpuppa) [inline zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerAuthorizationCreateRequest.java:25]: ❓ Why this is changed ? The class name is a bit confusing

Comment (nicpuppa) [reply zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerAuthorizationCreateRequest.java:25]: Nevermind, I saw the other PR

File: service/src/main/java/io/camunda/service/UserServices.java (hunk 2 above)
Comment (remcowesterhoud) [reply service/src/main/java/io/camunda/service/UserServices.java:48]: That's just my lack of knowledge of the REST API. I copied it from another request 😄 I'll change it.

**Review by nicpuppa — APPROVED** [review]
Thanks @remcowesterhoud
